onnxruntime/core/optimizer/qdq_transformer/qdq_propagation.cc

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "qdq_propagation.h"
+
+#include <deque>
+
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/utils.h"
+
+namespace onnxruntime {
+
+static constexpr size_t QDQInputCountRequired = 3;
+static constexpr size_t QDQInputScaleIdx = 1;
+static constexpr size_t QDQInputZeroPointIdex = 2;
+
+static bool CanNodePropagate(const Node& node) {
```

Comment:
Is is certain that CanPropagate backward()  conceptually equal to CanPropagate  forward()?

Reply:
No, just share them here. Will split them when adding more.